File: umbrella-insurance-react-ui/src/models/users/sessions/v1/Session.ts
import { Device } from "../../../devices/v1/Device";
import { User } from "../../v1/User";

export class Session {
    id?: number;
    user?: User;
    sessionCode?: string;
    startDateTime?: string;
    endDateTime?: string;
    minutesToExpire?: number;
    device?: Device;

    constructor(obj?: any) {
        if(obj) {
            if (obj?.id) {
                this.id = obj.id;
            }
            if (obj?.user) {
                this.user = new User(obj.user)
            }
            if (obj?.sessionCode) {
                this.sessionCode = obj.sessionCode;
            }
            if (obj?.startDateTime) {
                this.startDateTime = obj.startDateTime;
            }
            if (obj?.endDateTime) {
                this.endDateTime = obj.endDateTime;
            }
            if (obj?.minutesToExpire) {
                this.minutesToExpire = obj.minutesToExpire;
            }
            if (obj?.device) {
                this.device = new Device(obj.device);
            }
        }
    }
    /*
    toJSON() {
        return {
            "id": this.id ? this.id?.toString() : null,
            "user": this.user ? this.user?.toString() : null,
            "sessionCode": this.sessionCode ? this.sessionCode  : null,
            "startDateTime": this.startDateTime ? this.startDateTime : null,
            "endDateTime": this.endDateTime ? this.endDateTime : null,
            "minutesToExpire": this.minutesToExpire ? this.minutesToExpire?.toString() : null,
            "device": this.device ? this.device?.toString(): null
        }
    }
        */

}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Session.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "sessions", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "sessions_unique", columnNames = {"session_code"})
})
public class Session {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "session_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "session_code", length = Integer.MAX_VALUE)
    private String sessionCode;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "start_date_time", nullable = false)
    private Timestamp startDateTime;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "end_date_time", nullable = false)
    private Timestamp endDateTime;

    @NotNull
    @Column(name = "minutes_to_expire", nullable = false)
    private Long minutesToExpire;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "device_id")
    private Device device;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "verification_code_id")
    private VerificationCode verificationCode;

    @NotNull
    @Column(name = "did_sign_out", nullable = false)
    private Boolean didSignOut = false;

}